Drop Apache 2.2 support

ekohl opened the pull request:
This is likely incomplete and untested at the moment. However, it shows how big the cleanup is.

ekohl commented on manifests/default_mods.pp:
```diff
@@ -14,19 +13,13 @@
-          ::apache::mod { 'systemd': }
-        }
-        ::apache::mod { 'unixd': }
+      if $facts['os']['name'] != 'Amazon' && $use_systemd {
+        ::apache::mod { 'systemd': }
       }
+      if ($facts['os']['name'] == 'Amazon' and $facts['os']['release']['full'] == '2') {
+        ::apache::mod { 'systemd': }
+      }
```
Here I wonder what the best approach is. Why doesn't Amazon Linux support the `use_systemd` flag? And I think Amazon Linux 2022 is also out, which also has systemd. Should we simplify? Related: in other places I dropped Amazon Linux 1 support since it's based on EL6.

ekohl commented:
This is now green.

david22swan commented:
@ekohl This look's good to me so far, haven't finished the review yet though.
Will need to wait a bit on the merge though, give some time to get any other PRs that are near completion in a merged so we can cut a pre-major release, followed by some work to get any other backwards incompatible changes that we want in.

david22swan commented:
@ekohl As a not, there was previously work done by a former member of our team to remove 2.2 support from the module. While it was unable to be merged at the time due to push back from customers and the community I have made sure to keep the branch saved.
It is currently pushed up as 2.2removal on my fork. Was planning to work on rebasing it when I had the free time but if you want to use it as a help with this PR here it may be off assistance.

ekohl commented:
I'm pretty occupied for the next week or 2 so feel free to use my code.

david22swan commented:
@ekohl From how you had responded had thought that this needed more work to get it over the line, but looking now it seems to be complete to me. In your opinion is there anything that needs one on it other than rebasing?

ekohl commented:
Since https://github.com/puppetlabs/puppetlabs-apache/pull/2329#issuecomment-1289803115 I think this is ready. I've just rebased this.

david22swan commented:
Ok, will give it one last review and make a clean release before merging it in.

Apologies for my misunderstanding of the situation

ekohl commented:
No worries, next time I'll be more explicit. I do use the draft feature quite explicitly to mark my work as in progress.

ekohl commented:
No, it's because I only partially addressed #2331. The Puppet code gave a merge conflict so I addressed that and fixed up the template as well, but forgot the tests.

david22swan commented on manifests/version.pp:
```diff
@@ -6,37 +6,4 @@
   Optional[String] $scl_httpd_version = undef,
   Optional[String] $scl_php_version   = undef,
 ) {
-  case $facts['os']['family'] {
```
Seeing as how everything has been removed should this not ust be deleted?

ekohl replied:
I wonder. It does still have class parameters that can be used. Those could also be added to `params.pp`, but I don't know what's best here.

david22swan replied:
It's a hard question. I guess for now it may be best to leave them in place, at least until hiera is properly implemented within the module. Just so we have a clear place to set them.

david22swan reviewed (approved):
LGTM
Will merge once I can get a clean pre-major release out

chelnak commented:
This is awesome ❤️

david22swan commented:
Well, that's that
Thank you for all the work :)
